Sources/Client/Client/Client+Events.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  Client+Events.swift
+//  StreamChatClient
+//
+//  Created by Bahadir Oncel on 30.03.2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// Reference for the subscription initiated. Call `cancel()` to end subscription.
+public protocol Cancellable {
+    /// Cancel the underlying subscription for this object.
+    func cancel()
+}
+
+/// Reference for the subscription initiated. Call `cancel()` to end the subscription. Alternatively, when this object is deallocated, it calls `cancel()` on itself automatically.
+public protocol AutoCancellable: Cancellable {}
+
+struct Subscription: Cancellable {
+    private let onCancel: (String) -> ()
+    let uuid: String
+    
+    init(onCancel: @escaping (String) -> ()) {
+        self.onCancel = onCancel
+        uuid = UUID().uuidString
+    }
+    
+    public func cancel() {
+        onCancel(uuid)
+    }
+}
+
+class SubscriptionBag: Cancellable {
+    private var subscriptions = [Cancellable]()
+    
+    public func add(_ subscription: Cancellable)  {
+        subscriptions.append(subscription)
+    }
+    
+    public func cancel() {
+        subscriptions.forEach { $0.cancel() }
+    }
+}
+
+extension Client {
+    /// Observe events for the given event types.
+    /// - Parameters:
+    ///   - eventTypes: A set of event types to be observed. Defaults to all events.
+    ///   - callback: Callback closure to be called for each new event.
+    /// - Returns: `Subscription` object to be able to cancel observing. Call `subscription.cancel()` when you want to stop observing.
+    /// - Warning: Subscriptions do not cancel on `deinit` and that can cause crashes / memory leaks, so make sure you handle subscriptions correctly.
+    public func subscribe(forEvents eventTypes: Set<EventType> = Set(EventType.allCases), _ callback: @escaping OnEvent) -> Cancellable {
```

Comment:
You said you'll rename it to:
```suggestion
    public func subscribe(for eventTypes: Set<EventType> = Set(EventType.allCases), _ callback: @escaping OnEvent) -> Cancellable {
```

Reply:
I've renamed others (starting with `events` in Rx subset), it doesn't make sense for this func:
```swift
subscribe(forEvents: [.userPresenceChanged, .userUpdated]...)
// makes more sense then
subscribe(for: [.userPresenceChanged, .userUpdated]...)
```
what do you think?

Reply:
It depends. Just be try to be consistent everywhere and our SDK will look more solid.